pkg/monitortestframework/impl.go

```diff
@@ -310,3 +318,36 @@ func (r *monitorTestRegistry) AddRegistryOrDie(registry MonitorTestRegistry) {
 func (r *monitorTestRegistry) getMonitorTests() map[string]*monitorTesttItem {
 	return r.monitorTests
 }
+
+// GetOpenshiftTestsImagePullSpec returns the pull spec, a not-supported-reason, or an error.
+func (m *MonitorTestInitializationInfo) GetOpenshiftTestsImagePullSpec(ctx context.Context, adminRESTConfig *rest.Config) (string, string, error) {
+	payloadImagePullSpec := m.UpgradeTargetPayloadImagePullSpec
+	if len(payloadImagePullSpec) == 0 {
+		configClient, err := configclient.NewForConfig(adminRESTConfig)
+		if err != nil {
+			return "", "", err
+		}
+		clusterVersion, err := configClient.ConfigV1().ClusterVersions().Get(ctx, "version", metav1.GetOptions{})
+		if apierrors.IsNotFound(err) {
+			return "", "clusterversion/version not found and no image pull spec specified", nil
+		}
+		if err != nil {
+			return "", "", err
+		}
+		payloadImagePullSpec = clusterVersion.Status.History[0].Image
+	}
+
+	// runImageExtract extracts src from specified image to dst
+	cmd := exec.Command("oc", "adm", "release", "info", payloadImagePullSpec, "--image-for=tests")
```

Comment:
> I'm concerned that this will not work on disconnected/proxied environments

It won't, hence the soft handling.  If we don't do this, then it doesn't work on PRs.